lib/keystore.js

```diff
@@ -102,14 +106,47 @@ var KeyStore = function(mnemonic, pwDerivedKey, hdPathString) {
     // for standard Ethereum accounts.
 
     var hdRoot = new Mnemonic(mnemonic).toHDPrivateKey().xprivkey;
-    this.encHdRootPriv = KeyStore._encryptString(hdRoot, pwDerivedKey);
+    this.encHdRootPriv = encryptString(hdRoot, pwDerivedKey);
 
     var hdRootKey = new bitcore.HDPrivateKey(hdRoot);
     var hdPath = hdRootKey.derive(hdPathString).xprivkey;
-    pathKsData.encHdPathPriv = KeyStore._encryptString(hdPath, pwDerivedKey);
+    pathKsData.encHdPathPriv = encryptString(hdPath, pwDerivedKey);
+  }
+}
+
+KeyStore.createVault = function(opts, cb) {
+  var _this = this;
+
+  // Default hdPathString
+  if (!('hdPathString' in opts)) {
+    opts.hdPathString = "m/0'/0'/0'";
   }
+
+  // Default seed phrase if not specified
+  if (!('seedPhrase' in opts)) {
+    opts.seedPhrase = this.generateRandomSeed();
+  }
+
+  if (!('salt' in opts)) {
+    opts.salt = defaultSalt;
+  }
+
+  this.deriveKeyFromPassword(opts.password, opts.salt, function(err, pwDerivedKey) {
+    if (err) return cb(err);
+
+    var ks = new _this(opts.seedPhrase, pwDerivedKey, opts.hdPathString);
+    ks.init(opts.seedPhrase, pwDerivedKey, opts.hdPathString, opts.salt);
+
+    cb(null, ks);
+  });
 };
 
+KeyStore.generateSalt = generateSalt;
+
+function generateSalt (byteCount) {
+  return bitcore.crypto.Random.getRandomBuffer(byteCount || 12).toString('base64');
```

Comment:
Could be any length, had no reason to pick 12, if you think 32 is a better salt, let's do it.